### ISSUE TITLE
Panic when using an enum struct-like constructor through type alias

### ISSUE THREAD

**Issue by tchajed** [verus-lang/verus#401]
Verus panics when trying to do this (ignore the names, they are an artifact of minimizing an artificial example):

typs.rs:
```
use crate::pervasive::prelude::*;

verus! {
  #[is_variant]
  pub enum Direction {
    Left{},
  }

  pub type Node = Direction;
}
```

main.rs:
```
mod pervasive;
use pervasive::prelude::*;

mod typs;

verus! {
  use typs::*;

  fn mk_node() -> Node {
    Node::Left{}
  }

  fn main() {}
}
```

The part that doesn't work is `Node::Left{}`, which results in the following panic:
```
thread 'rustc' panicked at 'unexpected qpath TypeRelative(Ty { hir_id: HirId { owner: DefId(0:2099 ~ main[8185]::mk_node), local_id: 2 }, kind: Path(Resolved(None, Path { span: src/main.rs:10:5: 10:9 (#0), res: Def(TyAlias, DefId(0:2093 ~ main[8185]::typs::Node)), segments: [PathSegment { ident: Node#0, hir_id: Some(HirId { owner: DefId(0:2099 ~ main[8185]::mk_node), local_id: 1 }), res: Some(Err), args: None, infer_args: true }] })), span: src/main.rs:10:5: 10:9 (#0) }, PathSegment { ident: Left#0, hir_id: Some(HirId { owner: DefId(0:2099 ~ main[8185]::mk_node), local_id: 3 }), res: Some(Err), args: None, infer_args: true })', rust_verify/src/rust_to_vir_expr.rs:2878:22
```

This produces a good error message if `Direction::Left` is declared as `Left,` or `Left(),`.

**Comment by utaal** [comment]
I suspected this may happen. I think we will soon be overhauling naming to support multiple impls per (trait, type constructor); they're currently disallowed as they would have the same name with our current strategy. We will probably want canonical names for things like this too.

**Comment by tjhance** [comment]
i think this should work if we resolve through rustc_middle instead of just reading the AST path when handling the constructor.

@utaal I'm not sure if I'm missing something, but I don't think this issue has anything to do with multiple impls.

**Comment by tjhance** [comment]
fixed by 5397e0b60bddf90b2404d33563703e6594e597d5
